### PR TITLE
Imports UI changes from comp/2024.

### PR DESCRIPTION
**Pull request by barulicm**

### CODE REVIEW

File: ateam_ui/src/src/components/StatusComponent.vue [hunk 1]
```diff
@@ -144,12 +144,12 @@ export default {
...
-                textY: [-.12, -.12, .14, .14, -.13]
+                startX: [-.075, -.035, .035, .075, -.05],
+                startY: [-.095, .12, .12, -.095, -.09],
+                endX: [-.12, -.1, .1, .12, .05],
+                endY: [-.03, .07, .07, -.03, -.09],
+                textX: [-.12, -.12, .12, .12, 0],
+                textY: [-.12, .14, .14, -.12, -.13]
             }
 
             // TODO: Figure out what order the motor numbers use
```
Comment (gtrosawa) [inline ateam_ui/src/src/components/StatusComponent.vue:155]: 0: FL
1: BL
2: BR
3: FR
4: DRB

File: ateam_ui/src/src/components/StatusComponent.vue [hunk 2]
```diff
@@ -171,6 +171,7 @@ export default {
                     if (general)  errString = errString + "G";
                     if (hall)  errString = errString + "H";
                     if (encoder)  errString = errString + "E";
+                    errString = String(i);
```
Comment (gtrosawa) [inline ateam_ui/src/src/components/StatusComponent.vue:174]: Is this line intentional?
It seems like the error info will be stripped and replaced with the current index

File: ateam_ui/src/src/state.ts [hunk 3]
```diff
@@ -7,6 +7,7 @@ import { Ball } from "@/ball"
 import { Field, FieldDimensions, FieldSidedInfo } from "@/field"
 import { AIState } from "@/AI"
 import { Play } from "@/play"
+import { exportDefaultSpecifier } from "@babel/types"
```
Comment (gtrosawa) [inline ateam_ui/src/src/state.ts:10]: Seems like this is an unused import

File: ateam_ui/src/src/state.ts [hunk 4]
```diff
@@ -128,6 +135,43 @@ export class AppState {
...
+            int_side = -1;
+        }
+        const request = new ROSLIB.ServiceRequest({
+            ignore_side: int_side
+        });
+
+        this.services["setIgnoreFieldSide"].callService(request,
+            function(result) {
+                if(!result.success) {
+                    console.log("Failed to set ignore side: ", result.reason);
```
Comment (gtrosawa) [inline ateam_ui/src/src/state.ts:154]: Should probably be `console.error` or `console.warn`
Same with the other instances of logging unsuccessful requests

**Review by gtrosawa — COMMENTED** [review]
Mostly minor details that do not strictly require any action

Some questions on intent (e.g. overwriting an error code accumulator string before using its value somewhere) left as code comments

If they were indeed intended changes, I can go ahead and approve

File: ateam_ui/src/src/components/FieldComponent.vue [hunk 5]
```diff
@@ -1,58 +1,286 @@
...
+
+            const simulator_command = {
+                teleport_ball: [{
+                    pose: this.state.world.ball.pose
+                }],
+            };
+
+            this.state.sendSimulatorControlPacket(simulator_command);
+        },
+        rightClickDrag: function(event) {
```
[the remark this answers is not in the document]
Comment (barulicm) [reply ateam_ui/src/src/components/FieldComponent.vue:128]: This function just draws things so we can see what we're about to do. Line 69 registers an event handler for when the right click button is released, which actually sends the command to the simulator to move the ball and set its velocity.

File: ateam_ui/src/src/state.ts (hunk 3 above)
Comment (barulicm) [reply ateam_ui/src/src/state.ts:10]: vscode agrees. I'll remove it.

File: ateam_ui/src/src/components/StatusComponent.vue (hunk 1 above)
Comment (barulicm) [reply ateam_ui/src/src/components/StatusComponent.vue:155]: The wheels dict appears to have this order correct, so I'm guessing this TODO is just leftover and I'm going to remove it.

Comment (barulicm) [reply ateam_ui/src/src/components/StatusComponent.vue:155]: The wheels struct above seems to have them in the right order, so I'm guessing this TODO is leftover and I'll remove it.

File: ateam_ui/src/src/state.ts [hunk 6]
```diff
@@ -128,6 +135,43 @@ export class AppState {
             });
     }
 
+    setIgnoreFieldSide(ignore_side: number) {
+        const state = this; // fix dumb javascript things
```
[the remark this answers is not in the document]
Comment (barulicm) [reply ateam_ui/src/src/state.ts]: Removed.

File: ateam_ui/src/src/components/FieldComponent.vue [hunk 7]
```diff
@@ -1,58 +1,286 @@
...
+            const viewport = this.pixi.stage.getChildAt(0);
+            const ball_pos = viewport.getChildByName("ball").getChildAt(0).position;
```
[the remark this answers is not in the document]
Comment (barulicm) [reply ateam_ui/src/src/components/FieldComponent.vue]: Removed.

File: ateam_ui/src/src/components/StatusComponent.vue (hunk 2 above)
Comment (barulicm) [reply ateam_ui/src/src/components/StatusComponent.vue]: Moved the motor index to the start of the error character sequence and resolved this in the process.

**Review by gtrosawa — APPROVED** [review]

**Pull request merged**